spring-ai-docs/src/main/antora/modules/ROOT/pages/api/embeddings/postgresml-embeddings.adoc

```diff
@@ -0,0 +1,150 @@
+= PostgresML Embeddings
+
+Spring AI supports the PostgresML text embeddings models.
+
+Embeddings are a numeric representation of text.
+They are used to represent words and sentences as vectors, an array of numbers.
+Embeddings can be used to find similar pieces of text, by comparing the similarity of the numeric vectors using a distance measure, or they can be used as input features for other machine learning models, since most algorithms can't use text directly.
+
+Many pretrained LLMs can be used to generate embeddings from text within PostgresML.
+You can browse all the https://huggingface.co/models?library=sentence-transformers[models] available to find the best solution on Hugging Face.
+
+== Getting Started
+
+=== Configure the PostgresML Embeddings Client Manually
+
+Add the `spring-ai-postgresml` dependency to your project's Maven `pom.xml` file:
+[source, xml]
+----
+<dependency>
+    <groupId>org.springframework.ai</groupId>
+    <artifactId>spring-ai-postgresml</artifactId>
+    <version>0.8.0-SNAPSHOT</version>
+</dependency>
+----
+
+or to your Gradle `build.gradle` build file.
+
+[source,groovy]
+----
+dependencies {
+    implementation 'org.springframework.ai:spring-ai-postgresml:0.8.0-SNAPSHOT'
+}
+----
+
+Next, create an `PostgresMlEmbeddingClient` instance and use it to compute the similarity between two input texts:
+
+[source,java]
+----
+var jdbcTemplate = new JdbcTemplate(dataSource); // your posgresml data source
+
+PostgresMlEmbeddingClient embeddingClient = new PostgresMlEmbeddingClient(this.jdbcTemplate,
+        PostgresMlEmbeddingOptions.builder()
+            .withTransformer("distilbert-base-uncased") // huggingface transformer model name.
+            .withVectorType(VectorType.PG_VECTOR) //vector type in PostgreSQL.
+            .withKwargs(Map.of("device", "cpu")) // optional arguments.
+            .withMetadataMode(MetadataMode.EMBED) // Document metadata mode.
+            .build());
+
+embeddingClient.afterPropertiesSet(); // initialize the jdbc template and database.
```

Comment:
It would be helpful to note that `PostgresMlEmbeddingClient` bean is defined using `@Bean`, `afterPropertiesSet` is automatically called, so explicitly invoking it is not necessary. This clarification could prevent misunderstandings.

Reply:
absolutely